OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -330,6 +330,39 @@ public void SetLineWidth(float width)
 			ErrorHandler.CheckGlError();
 		}
 
+		public Bitmap TakeScreenshot()
+		{
+			var rect = new Rectangle(Point.Empty, size);
+			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var data = bitmap.LockBits(rect,
+				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+
+			try
+			{
+				int packAlignment;
+
+				unsafe
+				{
+					GL.GetInteger(GetPName.PackAlignment, out packAlignment);
+				}
+
+				GL.PixelStore(PixelStoreParameter.PackAlignment, 1);
```

Comment:
You should also set PixelStoreParameter.RowLength to the actual bitmap row stride given to you in `data`.